src/main/java/com/artipie/asto/memory/BenchmarkStorage.java

```diff
@@ -136,7 +138,28 @@ public BenchmarkStorage(final InMemoryStorage backend) {
 
     @Override
     public CompletableFuture<Void> move(final Key source, final Key destination) {
-        throw new NotImplementedError("Not implemented yet");
+        final CompletionStage<Void> res;
+        if (this.deleted.contains(source)) {
+            res = artipieIOcompletion("No value for source key", source);
+        } else {
+            final byte[] lcl = this.local.get(source);
```

Comment:
You can use `this.local.computeIfAbsent(source, key -> this.backend.data.get(key.string())` instead of these three lines